### PR TITLE
Added gencerts command, fixed unreachable code, added missing argument, fixed few golint errors

### LINKED ISSUE #868: Programmatically generate ssl certs
Create a `gencerts` command (cmd/gencerts.go) that will automatically generate self signed SSL certificates that can be used by the OpenBazaar electron client running on a remote machine. This is to make it easier for users to set up an instance on a remote machine and avoid manually typing openssl commands.

I gave it a go [here](https://github.com/OpenBazaar/openbazaar-go/pull/708) but could not get the client to accept the generated cert. 

@chaintip $50

### PR DESCRIPTION
**Pull request by trigun117**
@chaintip fixes #868

### CODE REVIEW

**Comment by chaintip** [comment]
If this pull request is merged to fix issue [#868](http://github.chaintip.org/OpenBazaar/openbazaar-go/issues/868) first, @trigun117 will claim the `0.03317627 BCH`| `~  54.35 USD` [bounty](http://github.chaintip.org/chaintip/bounties/issues/15) and all future tips to the issue.

File: cmd/setapicreds.go [hunk 1]
```diff
@@ -86,7 +88,7 @@ func (x *SetAPICreds) Execute(args []string) error {
 		apiCfg.AllowedIPs = []string{}
 	}
 
-	if err := r.SetConfigKey("JSON-API", apiCfg); err != nil {
+	if r.SetConfigKey("JSON-API", apiCfg); err != nil {
```
Comment (placer14) [inline cmd/setapicreds.go]: This does not look right. Which `err` did you mean to check here? 

I'm thinking the linter told you that you can't declare an `err` which has already been declared. But when you remove the `:` it tells you that `err` has not be declared inside the block. If this is the case, I've solved this by declaring the `err` at the top and using `=` throughout.

File: cmd/gencerts.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		return &k.PublicKey
+	default:
+		return nil
+	}
+}
+
+func pemBlockForKey(priv interface{}) *pem.Block {
+	switch k := priv.(type) {
+	case *rsa.PrivateKey:
+		return &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(k)}
```
Comment (placer14) [inline cmd/gencerts.go]: The preamble `"RSA PRIVATE KEY"` should probably be a constant.

File: cmd/gencerts.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+	// Set repo path
+	repoPath, err := repo.GetRepoPath(x.Testnet)
+	if err != nil {
+		return err
+	}
+	if x.DataDir != "" {
+		repoPath = x.DataDir
+	}
+
+	flag.Parse()
```
Comment (placer14) [inline cmd/gencerts.go]: Is this being called early enough? Seems `x.Testnet` would always be `false` here.

File: cmd/gencerts.go [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+	for _, h := range hosts {
+		if ip := net.ParseIP(h); ip != nil {
+			template.IPAddresses = append(template.IPAddresses, ip)
+		} else {
+			template.DNSNames = append(template.DNSNames, h)
+		}
+	}
+
+	template.IsCA = true
+	template.KeyUsage |= x509.KeyUsageCertSign
```
Comment (placer14) [inline cmd/gencerts.go]: Why are these floating down here like this? Why not add them to the `template` declaration on L89?

File: cmd/gencerts.go [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+
+	//Check if host ip or dns name and count their quantity
+	hosts := strings.Split(x.Host, ",")
+	for _, h := range hosts {
+		if ip := net.ParseIP(h); ip != nil {
+			template.IPAddresses = append(template.IPAddresses, ip)
+		} else {
+			template.DNSNames = append(template.DNSNames, h)
+		}
+	}
```
Comment (placer14) [inline cmd/gencerts.go]: Build your `[]string` for IPAddresses and DNSNames first, then assign it to the struct on L89 all at once.

File: cmd/restore.go [hunk 6]
```diff
@@ -47,6 +47,7 @@ import (
 	"time"
 )
 
+//Restore struct
```
Comment (placer14) [inline cmd/restore.go]: Adding empty definitions merely to satisfy the linter defeats the purpose of the linter. Let's remove these or replace them w idiomatic go docs. https://blog.golang.org/godoc-documenting-go-code

File: core/net.go [hunk 7]
```diff
@@ -402,7 +402,6 @@ func (n *OpenBazaarNode) SendDisputeClose(peerId string, k *libp2p.PubKey, resol
 		Payload:     a,
 	}
 	return n.sendMessage(peerId, k, m)
-	return nil
```
Comment (placer14) [inline core/net.go]: Thanks.

File: net/retriever/retriever.go [hunk 8]
```diff
@@ -388,7 +388,7 @@ func (m *MessageRetriever) handleMessage(env pb.Envelope, addr string, id *peer.
 					log.Errorf("Error saving offline message %s to database: %s", addr, err.Error())
 				}
 			} else {
-				log.Errorf("Error serializing offline message %s for storage")
+				log.Errorf("Error serializing offline message %s for storage", addr)
```
Comment (placer14) [inline net/retriever/retriever.go]: Thanks.

**Comment by chaintip** [comment]
Nice work @trigun117! Your pull request has been merged and the [bounty](http://github.chaintip.org/chaintip/bounties/issues/15) of `0.03317627 BCH`| `~  38.60 USD` has been sent to your wallet.

**Pull request merged**